File: sangsoo/0x0C Backtracking/15652.py
N, M = map(int, input().split())

arr = [0 for _ in range(M)]
isused = [0 for _ in range(N)]

def func(k : int, st: int):
    if(k == M):
        for i in range(M):
            print(arr[i], end = ' ');
        print("")
        return
    for i in range(st, N):
        arr[k] = i+1
        func(k+1, i)

func(0, 0)

File: sangsoo/0x0C Backtracking/15652.cpp
#include <iostream>
using namespace std;

int N, M;
int arr[10];
int isused[10];

void func(int k, int st){
    if(k == M){
        for(int i = 0; i < M; i++)
            cout << arr[i] << ' ';
        cout << '\n';
        return; // return 꼭 넣어줘야 함        
    }
    for(int i = st; i < N; i++){
        arr[k] = i + 1;
        func(k+1, i);
    }
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> N >> M ;
    func(0, 0);
}